gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentValues.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+
+/** Contains details about one content object: key, the live values and the expired values. */
+public class ContentValues implements Serializable {
+
+  private static final long serialVersionUID = -1268447510930923804L;
+  private ContentKey key;
+  private final Set<Content> expiredContents = new HashSet<>();
+  private final Set<Content> liveContents = new HashSet<>();
+
+  public ContentKey getKey() {
+    return key;
+  }
+
+  private void setIfAbsent(ContentKey key) {
+    if (this.key == null) {
```

Comment:
key is the content key (db + table name). Reference may not matter to the consumer as it is Nessie concept. consumer might care about what what table these contents belong. So that cleanup operations can be called on that key.

Reply:
But what does the table name mean without a reference name?

Reply:
For example, Iceberg expire snapshot is a consumer of GC results. It doesn't care what reference table belongs. It just needs the table name to call the expire snapshots. Hence not keeping the reference name in the output.

Reply:
And what happens when the table's renamed? Or when completely different tables tables have the same name in different references? Or when a table's being recreated with the same name?

Reply:
keeping reference name still may not help the consumer of GC in this situation if we don't have content id to reference mapping (including the dropped tables).
So we need to introduce this mapping? Any suggestions?

Reply:
**Agree, we need reference info also**. 
Initially I was thinking as the global state is common, only once I can call expire snapshot per content id (table) on default reference.
But the table can be renamed or may not exist in default reference. So, I have to call expire snapshots for table@reference. Internally I have to skip passing the reference which are cleaned up already by checking against global state's snapshots list.

Reply:
And I cannot merge the snapshot ids for per content id. I have to keep separately per reference. 
I need to change
`content id -> content key + list of all the snapshots from all the reference.`
to 
`refName -> content id -> content key +  list of expired snapshots of this reference.`